modules/Microsoft.Compute/virtualMachines/deploy.bicep

```diff
@@ -466,6 +466,7 @@ resource vm 'Microsoft.Compute/virtualMachines@2021-07-01' = {
           deleteOption: contains(nicConfiguration, 'deleteOption') ? nicConfiguration.deleteOption : 'Delete'
           primary: index == 0 ? true : false
         }
+        #disable-next-line use-resource-id-functions
```

Comment:
There's something we need to double check about the implementation of networkProfile in VM vs the one in VMSS. In VMSS we implement that differently and we don't have this warning. I'd suggest to remove this workaround to have this failing and try to solve it instead.
```suggestion
```